test/58-assert-11-assert_not_equal_files.bats

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env bats
+
+load 'test_helper'
+fixtures 'exist'
+
+# Correctness
+@test 'assert_not_equal_files() <file1> <file2>: returns 0 if <file1> and <file2> are not the same' {
+  local -r file1="${TEST_FIXTURE_ROOT}/dir/file_with_text"
+  local -r file2="${TEST_FIXTURE_ROOT}/dir/file"
+  run assert_not_equal_files "$file1" "$file2"
+  [ "$status" -eq 0 ]
+  [ "${#lines[@]}" -eq 0 ]
+}
+
+
+@test 'assert_not_equal_files() <file1> <file2>: returns 1 if <file1> and <file2> are the same' {
+  local -r file1="${TEST_FIXTURE_ROOT}/dir/same_file_with_text"
+  local -r file2="${TEST_FIXTURE_ROOT}/dir/file_with_text"
+  run assert_not_equal_files "$file1" "$file2"
+  [ "$status" -eq 1 ]
+  [ "${#lines[@]}" -eq 3 ]
+  [ "${lines[0]}" == '-- files are the same --' ]
+  [ "${lines[1]}" == "path : $file" ]
+  [ "${lines[2]}" == '--' ]
+}
+
+# Transforming path
+@test 'assert_not_equal_files() <file1> <file2>: used <file2> as a directory' {
+  local -r file2="${TEST_FIXTURE_ROOT}/dir/same_file_with_text"
+  local -r file1="${TEST_FIXTURE_ROOT}/dir"
+  run assert_not_equal_files "$file1" "$file2"
+  [ "$status" -eq 0 ]
+  [ "${#lines[@]}" -eq 1 ]
+  [ "${lines[0]}" == "cmp: ${TEST_FIXTURE_ROOT}/dir: Is a directory" ]
+  [ "${lines[1]}" == "" ]
+  [ "${lines[2]}" == '' ]
+}
+
+@test 'assert_not_equal_files() <file1> <file2>: replace suffix of displayed path' {
```

Comment:
How these prefix, suffix, infix test generally should look like:
You set a specific path. By using `BATSLIB_FILE_PATH_REM` and `BATSLIB_FILE_PATH_ADD` you can remove and add parts to the path you have given. In order to test this correctly, you want your test to fail, so gives the paths as output and you can test if the paths have been changed with these ADD and REM variables as they should.

For example
```
  local -r BATSLIB_FILE_PATH_REM="#${TEST_FIXTURE_ROOT}"
```
This will remove at the beginning of your filepath `${TEST_FIXTURE_ROOT}` with % you would remove it at the end of the path `%${TEST_FIXTURE_ROOT}`
Then you add instead a `..`
```
  local -r BATSLIB_FILE_PATH_ADD='..'
```
When you then run your assert command:
```
  run assert_not_exist "${TEST_FIXTURE_ROOT}/dir/file"
```
Your path `"${TEST_FIXTURE_ROOT}/dir/file` will change to `"../dir/file`